### DISCUSSION TITLE
Heated seats/steering wheel controls?

### DISCUSSION THREAD

**Discussion by UnbridledGames** [alandtse/tesla#106] [Q&A]
I don't see any controls for seats or the steering wheel in the add-on. Is that because it's not possible (it should be - I can do it in the official app), or because it hasn't been sniffed/discovered/coded yet? 

It's winter here now, and my seat is freezing in the morning. My car auto-heats itself, but won't do the seat heaters. One of the main reasons I installed this add-on.

If it is possible, and just not implemented, I'm happy to try adding the functionality. If anyone could point me to any documentation on how to start looking for the info necessary, that'd be great. (I'm a programmer and I've reverse engineered apps/api's before. What I'm asking for is if there's any already written documentation so I don't have to start from scratch with the Tesla app)

**Comment by UnbridledGames** [comment]
Figured it out. Integrations already exist in teslajsonpy, they just haven't been added to tesla_custom. Forked the code, added a test steering wheel heater, and got it to work. Once I do some more with it and get the seat heaters going I'll try to clean it up enough for a pull request.

**Reply by alandtse** [reply]
Thanks. We've exposed it generally as a service, just no UI has been made since HA didn't have a good representation in the past.
https://github.com/alandtse/tesla/issues/10

There should be a select implemented in teslajsonpy that could be exposed now in HA. I'd suggest looking there.
https://github.com/zabuldon/teslajsonpy/pull/210
https://developers.home-assistant.io/docs/core/entity/select/

**Comment by UnbridledGames** [comment]
Is it exposed through the tesla_custom.api service (or something similar I’m not at my PC at the moment)? I tried using that and kept getting errors. Maybe I’m using it wrong. Didn’t see much guidance on its use however.

**Reply by alandtse** [reply]
Yes. It's literally calling the convenience function.
https://teslajsonpy.readthedocs.io/en/latest/teslajsonpy/teslajsonpy.html#teslajsonpy.Controller.api. There are already issues discussing some of the common issues with it. Please review those.
